docs/glossary.md

```diff
@@ -0,0 +1,35 @@
+# Corpus
+
+# Data schemas
+
+Different data structures we observe in the end to end data collection flow for Elastic (Beats & Agent). See [./data-schemas.md](./data-schemas.md).
+
+In the context of this tool it also refers to folder names containing template information for a specific data schema.
+
+# Dataset
+
+A dataset is a component of a [Data Stream](https://www.elastic.co/guide/en/fleet/master/data-streams.html). Is defined by an _integration_ and describes the ingested data and its structure.
+
+In the context of this tool `dataset` refers to the name of the dataset you can generate data for. Data structure and definitions are part of the integration package the dataset belongs to.
+
+# Fields definition
+
+Datasets define data structure. Within them there are fields definitions that describe the fields a dataset provides. 
+
+In the context of this tool we can refer to:
+- the fields definition within a dataset in an integration package
+- a file named `fields.yml` that contains fields definition that is not (yet) part of a package
+
+# Fields definition configuration
```

Comment:
I have to say: having `Fields definition` and `Fields definition configuration` is a little confusing for me :)

I'd prefer some more specific and different definition, to create less confusion, what about
`Fields definition` and `Fields configuration`?